generator/src/main/java/com/scottlogic/deg/generator/guice/CombinationStrategyProvider.java

```diff
@@ -0,0 +1,35 @@
+package com.scottlogic.deg.generator.guice;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.scottlogic.deg.generator.generation.GenerationConfig;
+import com.scottlogic.deg.generator.generation.combinationstrategies.*;
+
+public class CombinationStrategyProvider  implements Provider<CombinationStrategy> {
+    private final GenerationConfig config;
+
+    @Inject
+    public CombinationStrategyProvider(GenerationConfig config){
+        this.config = config;
+    }
+
+    @Override
+    public CombinationStrategy get() {
+        if (config.getDataGenerationType() == GenerationConfig.DataGenerationType.RANDOM){
+            return new MinimalCombinationStrategy();
```

Comment:
This should probably have some form of comment to explain why the minimal strategy is used for random.